docs/compatibility.md

```diff
@@ -0,0 +1,25 @@
+# Compatibility
```

Comment:
I think that a couple of hints in `CONTRIBUTING.md` mentioning `VERSION_MATCHES`, `FIXUPS`, and so forth might be beneficial for external developers.
Doesn't need to be a fully fledged guide, but a just enough list of hints.